doc/SAI-Proposal-VlanStack-QinQ.md

```diff
@@ -0,0 +1,380 @@
+
+#### Usage
+
+```
+sai_object_id_t vlan_stacking_oid;
+sai_attribute_t attr;
+vector<sai_attribute_t> vlan_stacking_entry_attrs;
+
+attr.id = SAI_VLAN_STACK_ATTR_STAGE;
+attr.value.s32 = SAI_VLAN_STACK_STAGE_INGRESS;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ACTION;
+attr.value.s32 = SAI_VLAN_STACK_ACTION_SWAP;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_MATCH_TYPE;
+attr.value.s32 = SAI_VLAN_STACK_MATCH_TYPE_INNER;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID_LIST;
+sai_uint16_t vid_list[] = { 10 }
+attr.value.u16list.list = vid_list;
+attr.value.u16list.count = 1;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_PORT;
+attr.value.oid = 10;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_APPLIED_VLAN_ID;
+attr.value.u16 = 100;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_VLAN_APPLIED_PRI;
+attr.value.u8 = 0;
+vlan_stacking_entry_attrs.push(attr);
+
+sai_vlan_api->create_vlan_stack(&vlan_stacking_oid, gSwitchId, (uint32_t)vlan_stacking_entry_attrs.size(), vlan_stacking_entry_attrs.data());
+```
+
+#### Behavior
+
+![vlan_stack-ingress-swap-inner](figures/vlan_stack/ingress_swap_inner.png "Table 4: Ingress Swap Inner")
+__Table 4: Vlan Stack Ingress Swap Inner__
+
+### Ingress/Swap outer tag based on outer tag
+Change outer dot1q header to VID=100 for those packets with original outer dot1q header VID=10 when they flow into Ethernet10
+
+#### Usage
+
+```
+sai_object_id_t vlan_stacking_oid;
+sai_attribute_t attr;
+vector<sai_attribute_t> vlan_stacking_entry_attrs;
+
+attr.id = SAI_VLAN_STACK_ATTR_STAGE;
+attr.value.s32 = SAI_VLAN_STACK_STAGE_INGRESS;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ACTION;
+attr.value.s32 = SAI_VLAN_STACK_ACTION_SWAP;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_MATCH_TYPE;
+attr.value.s32 = SAI_VLAN_STACK_MATCH_TYPE_OUTER;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID_LIST;
+sai_uint16_t vid_list[] = { 10 }
+attr.value.u16list.list = vid_list;
+attr.value.u16list.count = 1;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_PORT;
+attr.value.oid = 10;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_APPLIED_VLAN_ID;
+attr.value.u16 = 100;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_VLAN_APPLIED_PRI;
+attr.value.u8 = 0;
+vlan_stacking_entry_attrs.push(attr);
+
+sai_vlan_api->create_vlan_stack(&vlan_stacking_oid, gSwitchId, (uint32_t)vlan_stacking_entry_attrs.size(), vlan_stacking_entry_attrs.data());
+```
+
+#### Behavior
+
+![vlan_stack-ingress-swap-outer](figures/vlan_stack/ingress_swap_outer.png "Table 5: Ingress Swap Outer")
+__Table 5: Vlan Stack Ingress Swap Outer__
```

Comment:
Typo??
![image](https://user-images.githubusercontent.com/56053977/165821012-9467512a-170e-42d0-b321-9a57a47d5b77.png)
Here outer matches with VID 10 and gets swapped to VID100

Reply:
Yes, thank you. I will fix this in next commit.